File: src/main/java/org/zenbot/szolnok/timetable/batch/domain/Bus.java
package org.zenbot.szolnok.timetable.batch.domain;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Data
@Document
public class Bus {

    @Id
    private String id;
    private String busName;
    private List<BusRoute> busRoutes;

    public BusRoute getBusRouteByStartStopName(String startBusStopName) {
        Optional<BusRoute> routeLine = busRoutes.stream()
                .filter(line -> line.getStartBusStop().equals(startBusStopName))
                .findFirst();
        if (routeLine.isPresent()) {
            return routeLine.get();
        } else {
            BusRoute result = new BusRoute();
            result.setBusStops(new ArrayList<>());
            return result;
        }
    }

    public boolean hasNoBusRoute(BusRoute busRoute) {
        return !busRoutes.stream()
                .map(line -> line.getStartBusStop())
                .collect(Collectors.toList())
                .contains(busRoute.getStartBusStop());
    }
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/domain/Bus.kt
package org.zenbot.szolnok.timetable.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Bus(
    @Id var id: String = "",
    var busName: String = "",
    var busRoutes: MutableList<BusRoute> = ArrayList()
) {

    fun getBusRouteByStartStopName(startBusStopName: String): BusRoute {
        val routeLine = this.busRoutes.stream()
                .filter { line -> line.startBusStop == startBusStopName }
                .findFirst()
        if (routeLine.isPresent) {
            return routeLine.get()
        } else {
            val result = BusRoute()
            result.busStops = ArrayList()
            return result
        }
    }

    fun hasNoBusRoute(busRoute: BusRoute): Boolean {
        return this.busRoutes.stream()
                .map { line -> line.startBusStop }
                .filter { startStop -> busRoute.startBusStop == startStop }
                .count() == 0L
    }
}
